atest/robot/running/for.robot

```diff
@@ -254,6 +254,70 @@ For In Range With Multiple Variables
     Should Be For Item    ${tc.kws[0].kws[2]}    \${i} = 5, \${j} = 6, \${k} = 7
     Should Be For Item    ${tc.kws[0].kws[3]}    \${i} = 8, \${j} = 9, \${k} = 10
 
+For In Zip
+    ${tc} =    Check Test Case    ${TEST NAME}
+    ${for_loop}=    Set Variable    ${tc.kws[2]}
+    Should Be For In Zip Keyword    ${for_loop}    4
+    Should Be For Item    ${for_loop.kws[0]}    \${item} = a, \${thing} = e
+    Should Be For Item    ${for_loop.kws[1]}    \${item} = b, \${thing} = f
+    Should Be For Item    ${for_loop.kws[2]}    \${item} = c, \${thing} = g
+    Should Be For Item    ${for_loop.kws[3]}    \${item} = d, \${thing} = h
+
+For In Zip With Uneven Lists
+    [Documentation]    Handling lists with different number of elements. Probably should work like in Python here and just stop when the shortest list is empty?
+    ${tc} =    Check Test Case    ${TEST NAME}
+    ${for_loop}=    Set Variable    ${tc.kws[2]}
+    Should Be For In Zip Keyword    ${for_loop}    3
+    Should Be For Item    ${for_loop.kws[0]}    \${item} = a, \${thing} = d
+    Should Be For Item    ${for_loop.kws[1]}    \${item} = b, \${thing} = e
+    Should Be For Item    ${for_loop.kws[2]}    \${item} = c, \${thing} = f
+
+For In Zip With 3 Lists
+    [Documentation]    Handling more than one list like :FOR ${a} ${b} ${c} IN ZIP ${x} ${y} ${z}.
+    ${tc} =    Check Test Case    ${TEST NAME}
+    ${for_loop}=    Set Variable    ${tc.kws[3]}
+    Should Be For In Zip Keyword    ${for_loop}    4
+    Should Be For Item    ${for_loop.kws[0]}    \${item} = a, \${thing} = e, \${stuff} = 1
+    Should Be For Item    ${for_loop.kws[1]}    \${item} = b, \${thing} = f, \${stuff} = 2
+    Should Be For Item    ${for_loop.kws[2]}    \${item} = c, \${thing} = g, \${stuff} = 3
+    Should Be For Item    ${for_loop.kws[3]}    \${item} = d, \${thing} = h, \${stuff} = 4
+
+For In Zip With Other Iterables
+    [Documentation]    Handling non-lists. Should accept anything iterable except strings and fail with a clear error message if invalid data given. You can use utils.is_list_like to verify inputs.
+    Comment
+
+For In Zip With String "Lists"
+    [Documentation]    FAIL
+    ${tc} =    Check Test Case    ${TEST NAME}    status=FAIL    message=For-In-Zip Loop items must all be List-like (but not Strings); got <type 'unicode'> with value 'NotAListButStillIterable'
```

Comment:
Typically better to have expected status and message on the test data side on documentation. Specifying them explicitly is useful when you want to run same tests multiple times with different configuration and expected results.